backend/crowdin/events.go

```diff
@@ -0,0 +1,106 @@
+package crowdin
+
+import (
+	"context"
+	"github.com/ansel1/merry/v2"
+	"github.com/bcc-code/brunstadtv/backend/common"
+	"github.com/bcc-code/brunstadtv/backend/directus"
+	"github.com/bcc-code/brunstadtv/backend/events"
+	"github.com/bcc-code/mediabank-bridge/log"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/go-resty/resty/v2"
+	"github.com/samber/lo"
+)
+
+var CollectionNotSupported = merry.Sentinel("collection not supported")
+
+type PubSubEvent struct {
+	Type string `json:"type"`
+}
+
+type services interface {
+	GetDirectusClient() *resty.Client
+}
+
+type config interface {
+	GetCrowdinToken() string
+	GetCrowdinProjectIDs() []int
+}
+
+func HandleEvent(ctx context.Context, services services, config config, event cloudevents.Event) (err error) {
+	client := New(config.GetCrowdinToken(), ClientConfig{
+		ProjectIDs: config.GetCrowdinProjectIDs(),
+	})
+	switch event.Type() {
+	case events.TypeTranslationsSync:
+		handler := directus.NewHandler(services.GetDirectusClient())
+		return client.Sync(ctx, handler)
+	}
+	return merry.New("Unsupported event")
+}
+
+func toTranslationSources[t TranslationSource](items []t) []TranslationSource {
+	return lo.Map(items, func(i t, _ int) TranslationSource {
+		return i
+	})
+}
+
+func getStatusForItem(ctx context.Context, d *directus.Handler, collection string, id int) (status string, err error) {
+	var i hasStatus
+	switch collection {
+	case "shows":
+		i, err = d.GetShow(ctx, id)
+	case "seasons":
+		i, err = d.GetSeason(ctx, id)
+	case "episodes":
+		i, err = d.GetEpisode(ctx, id)
+	}
+	if err == nil {
+		status = i.GetStatus()
+	}
+	return
+}
+
+func getTranslationsForItem(ctx context.Context, d *directus.Handler, collection string, id int, language string) ([]TranslationSource, error) {
+	switch collection {
+	case "shows":
+		ts, err := d.ListShowTranslations(ctx, language, false, id)
+		if err != nil {
+			return nil, err
+		}
+		return toTranslationSources(ts), nil
+	case "seasons":
+		ts, err := d.ListSeasonTranslations(ctx, language, false, id)
+		if err != nil {
+			return nil, err
+		}
+		return toTranslationSources(ts), nil
+	case "episodes":
+		ts, err := d.ListEpisodeTranslations(ctx, language, false, id)
+		if err != nil {
+			return nil, err
+		}
+		return toTranslationSources(ts), nil
+	}
```

Comment:
```suggestion
        var err Error
        var ts <TYPE>
	case "shows":
		ts, err := d.ListShowTranslations(ctx, language, false, id)
	case "seasons":
		ts, err := d.ListSeasonTranslations(ctx, language, false, id)
	case "episodes":
		ts, err := d.ListEpisodeTranslations(ctx, language, false, id)
	}
	if err != nil {
			return nil, err
		}
		return toTranslationSources(ts), nil
```
Maybe?

Reply:
Need the type conversion. If there's a way to do this explicitly/implicitly another way, then that's nice, but I need to convert [](TranslationSource) to []TranslationSource